### ISSUE TITLE
Second call to openadas.repository.populate() function results in error and corrupted json file

### ISSUE THREAD

**Issue by vsnever** [cherab/core#243]
When cherab.openadas.repository.populate() function is called for the second time, namely when the local repository has already been created, this call throws the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vlad/Code/cherab-core/cherab/openadas/repository/create.py", line 382, in populate
    install_files(rates, download=download, repository_path=repository_path, adas_path=adas_path)
  File "/home/vlad/Code/cherab-core/cherab/openadas/install.py", line 39, in install_files
    install_adf11scd(*args, download=download, repository_path=repository_path, adas_path=adas_path)
  File "/home/vlad/Code/cherab-core/cherab/openadas/install.py", line 95, in install_adf11scd
    repository.update_ionisation_rates(rate_cherab, repository_path)
  File "/home/vlad/Code/cherab-core/cherab/openadas/repository/atomic.py", line 67, in update_ionisation_rates
    _update_and_write_adf11(species, rate_data, path)
  File "/home/vlad/Code/cherab-core/cherab/openadas/repository/atomic.py", line 203, in _update_and_write_adf11
    json.dump(content, f, indent=2, sort_keys=True)
  File "/usr/lib/python3.8/json/__init__.py", line 179, in dump
    for chunk in iterable:
  File "/usr/lib/python3.8/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.8/json/encoder.py", line 353, in _iterencode_dict
    items = sorted(dct.items())
TypeError: '<' not supported between instances of 'int' and 'str'
```
and also corrupts the .json file with ionization rate for hydrogen.

This happens because `charge` in `_update_and_write_adf11()` is not converted into str when updating the rate. So, instead:
```python
# update file content with new rate
        content[charge] = {
            'te': te.tolist(),
            'ne': ne.tolist(),
            'rate': rate_table.tolist(),
        }
```
must be:
```python
# update file content with new rate
        content[str(charge)] = {
            'te': te.tolist(),
            'ne': ne.tolist(),
            'rate': rate_table.tolist(),
        }
```
Also, the function `_update_and_write_adf11()` is duplicated in openadas/repository/radiated_power.py for some reason. I think it should be imported in radiated_power.py from openadas/repository/atomic.py.

I'll fix this bug.

**Comment by vsnever** [comment]
>Also, the function _update_and_write_adf11() is duplicated in openadas/repository/radiated_power.py for some reason. I think it should be imported in radiated_power.py from openadas/repository/atomic.py.

Another option is to move this function to utility.py and remove the '_' prefix. Otherwise, we are importing protected function from another module.

**Comment by CnlPepper** [comment]
If I remember correctly, the reason they are separate is because each file is meant to handle a separate part of the repository... and I think I was originally going to refactor the code but life changes meant it never happened. ADF11 shouldn't really be mentioned in the repo core as it is meant to be agnostic to the data source, however we never got around to implementing a more general representation of the rates. There is a lot of scope for cleanup and extension. To be honest, I'd just leave it until there is an effort to extend the system.

Amazed the str bug wasn't captured.... I thought I tested reinstalling multiple times! Nice stop.

**Comment by vsnever** [comment]
>Amazed the str bug wasn't captured.... I thought I tested reinstalling multiple times! Nice stop.

I also do not remember this error in Python 3.6. I caught this in Python 3.8.

> To be honest, I'd just leave it until there is an effort to extend the system.

Ok, then I'll just fix int to str conversion in atomic.py and radiated_power.py.

**Comment by vsnever** [comment]
Fixed in #244.
